query/src/main/java/org/infinispan/query/impl/massindex/DistributedExecutorMassIndexer.java

```diff
@@ -57,6 +61,86 @@ public void start() {
       return executeInternal(true);
    }
 
+   @Override
+   public CompletableFuture<Void> reindex(Object... keys) {
```

Comment:
This method is not doing anything wrong, but we can simplify it further:

This method is retrieving the value for each key, calculating the ```MassIndexStrategy``` for each value, and then calculating the mapping between keys and servers in order to route them to get indexed.

Instead it could calculate the ```MassIndexStrategy``` **once** for each **type**  in ```searchIntegrator.getIndexedTypes()``` without the need to retrieve the values.

After calculating the ```MassIndexStrategy```, it can either send the task (together with the keys) everywhere if ```IndexingExecutionMode.ALL``` or if ```IndexingExecutionMode.PRIMARY_ONWER``` it can use the method which will calculate the mapping between servers and keys for us:

```DistributedExecutorService.submitEverywhere(DistributedTask<T> task, K... input)```



So no need to manually create a ```Map<Address, List<Object>> addressKeysMap```.